RedRock.Calendar.Modules.Finance.Service/FinanceService.cs

```diff
@@ -21,23 +23,47 @@ public FinanceService(IFinanceBusinessLogic financeLogic, IEventService eventSer
             this.userService = userService;
         }
 
-        public async Task<FinanceDTO> GetMonthlyFee(Guid userReference, DateTime start, DateTime end)
+        public async Task<IEnumerable<FinanceDTO>> GetMonthlyFee(Guid userReference, DateTime start, DateTime end)
         {
+            var userRole = await userService.GetUserRole(userReference);
+            var financeDTOs = new List<FinanceDTO>();
+            if (userRole.Equals(UserRole.PropertyOwner))
```

Comment:
I would use the strategy pattern to return different values based on the user role

Reply:
or a factory